qiskit/pulse/compiler.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
```

Comment:
@ajavadia: any concern about `qiskit.compiler` and `qiskit.pulse.compiler`?

Reply:
I would prefer we put this all in either a unified compiler or something like [`Numba's approach`](https://github.com/numba/numba/tree/master/numba/core), here they have a `core`, `ir` and `compiler` and then other extensions derive their work from this such as [cuda](https://github.com/numba/numba/tree/master/numba/cuda). Something like this:
- core
    - ir
    - compiler
- circuit
    - ir
    - compiler
- pulse
   - ir
   - compiler

or something like:

- ir
    - core
    - classical
    - circuit
    - pulse
- compiler
     - frontend
     - optimization
     - lowering

I'm fine to put this wherever we agree we should.